### PR TITLE
store/tikv: refactor transaction committer.

### PR DESCRIPTION
**Pull request by coocood**
Replaced several arguments with a single action type, improves readability.

### CODE REVIEW

**Comment by coocood** [comment]
@disksing @tiancaiamao PTAL

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -26,7 +26,28 @@ import (
...
+	case actionCommit:
+		return "commit"
+	case actionCleanup:
+		return "cleanup"
+	}
+	return "unknown"
+}
+
+// twoPhaseCommitter executes a two-phase commit protocol.
+type twoPhaseCommitter struct {
```
Comment (shenli) [inline store/tikv/2pc.go]: Will there be a one phrase committer?

File: store/tikv/2pc.go [hunk 2]
```diff
@@ -318,32 +353,32 @@ func (c *txnCommitter) cleanupSingleRegion(bo *Backoffer, batch batchKeys) error
...
-		log.Errorf("txn failed cleanup key: %v, tid: %d", err, c.startTS)
+		err = errors.Errorf("2PC cleanup failed: %s", keyErr)
+		log.Errorf("2PC failed cleanup key: %v, tid: %d", err, c.startTS)
 		return errors.Trace(err)
 	}
 	return nil
 }
 
-func (c *txnCommitter) prewriteKeys(bo *Backoffer, keys [][]byte) error {
-	bo.ctx = context.WithValue(bo.ctx, cancelOnFirstError, struct{}{})
```
Comment (shenli) [inline store/tikv/2pc.go]: Why remove this line?

Comment (coocood) [inline store/tikv/2pc.go]: Since we have action parameter, we no longer need this to only cancel prewrite.

File: store/tikv/2pc.go (hunk 1 above)
Comment (coocood) [inline store/tikv/2pc.go]: It's 2PC protocol,  '2' is a digit, can't be used at first character.
We could do one-phase commit optimization for single region transaction in the future.

File: store/tikv/2pc.go [hunk 3]
```diff
@@ -89,82 +111,95 @@ func newTxnCommitter(txn *tikvTxn) (*txnCommitter, error) {
...
-		err = c.doBatches(bo, batches[:1], f)
+		err = c.doActionOnBatches(bo, action, batches[:1])
 		if err != nil {
 			return errors.Trace(err)
 		}
 		batches = batches[1:]
 	}
-	if asyncNonPrimary {
+	if action == actionCommit {
+		// commit secondary batches in background goroutine to reduce latency.
```
Comment (zimulala) [inline store/tikv/2pc.go]: s/commit/Commit

**Comment by tiancaiamao** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**